### DISCUSSION TITLE
org.apache.rocketmq.store.schedule.ScheduleMessageService.DeliverDelayedMessageTimerTask#messageTimeup

### DISCUSSION THREAD

**Discussion by EvanMi** [apache/rocketmq#7754] [General]
In this method , the code TopicFilterType topicFilterType = MessageExt.parseTopicFilterType(msgInner.getSysFlag()); means thre topicFilterType will always be SINGLE_TAG ? And why the sysFlag in MessageExt is ignored?

**Comment by leizhiyuan** [comment]
Here is a bitwise logic
